manifests/charts/gateways/istio-egress/templates/deployment.yaml

```diff
@@ -264,6 +268,10 @@ spec:
 {{ toYaml $gateway.additionalContainers | indent 8 }}
 {{- end }}
       volumes:
+      - emptyDir: {}
+        name: workload-socket
+      - emptyDir: {}
+        name: workload-certs
```

Comment:
should we only declare the volumes when spire agent is integrated.  Otherwise this is redundant

Reply:
Actually the `workload-socket` volume is always used, either by spire or by the istio-agent to mount its own SDS server. And as the integration is not configured through any property but it's based on the detection of either the socket or the certs file, we could not avoid creating the volumes.

Reply:
I think there's a lot of value to users in making it "just work" in environments with SPIRE due to the socket existing/not (like we make the certs-on-disk "just work" already). My preferred approach would be to keep this as-is and have it "just work".

However, we _could_ push this up to a top-level option in the install API - "useSPIRE" or something - to configure the mounts and ensure we don't get into the case @howardjohn mentioned (workload restarted and istio-agent didn't clean up the socket so it doesn't start its own SDS server and Envoy never gets a cert). That top level flag would configure the template mounts, but also control the logic in istio-agent's `Run` to decide how the SDS server is started. It's only worthwhile to do this if we can't agree on a clean-enough solution to the socket/SDS server lifecycle stuff.

Reply:
Is it correct to assume workload-socket and workload-certs can be used by non SPIRE env as well as long as it uses spiffe?  Or is this statement only for workload-socket?